### ISSUE TITLE
Add ignoreDefaultParamAssign option to space-infix-ops

### ISSUE THREAD

**Issue by kasvtv** [eslint/eslint#10398]
**What rule do you want to change?**
`space-infix-ops`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer!

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new option called `ignoreDefaultParamAssign` that will relax the rule when it comes to assigning defaults for function parameters.

**Please provide some example code that this change will affect:**

```js

// Currently not allowed:
function fun(value, multiplier=1, callback, setting=false) {
    value = 3+4;
    ....
}

// Currently allowed:
function fun(value, multiplier = 1, callback, setting = false) {
    value = 3 + 4;
    ....
}

// Not allowed with proposed option:
function fun(value, multiplier=1, callback, setting=false) {
    value = 3+4;
    ....
}

// Allowed with proposed option:
function fun(value, multiplier=1, callback, setting=false) {
    value = 3 + 4;
    ....
}



```

**What does the rule currently do for this code?**
A dev team could argue that assigning a single primitive value as a default for a function parameter would be more readable in a list of other parameters, because it couples the name and value together whilst separating it from the other variables by spaces.

**Comment by kasvtv** [comment]
Glad you're on board on this one!

And yes, we should camelCase the name of course! I happened to forget that the rule names with dashes and the options are camelCase.

**Comment by not-an-aardvark** [comment]
This was previously proposed in https://github.com/eslint/eslint/issues/3587.

Personally, I think it would be better to add a general option for the rule that allows customized spacing with [AST selectors](https://eslint.org/docs/developer-guide/selectors), rather than adding a specific option for default parameters. I'm not strongly opposed to adding an option for default parameters in the meantime (since we could deprecate it if we added a selector option later), but it seems like a selector option would cover all of the same cases. (For example, we would also be able to deprecate the existing `int32Hint` option.)

**Comment by kasvtv** [comment]
That sounds like the best choice. It's one of those settings that will pop up quite a bit if it's enabled, so finely tuned customization would be useful.

**Comment by platinumazure** [comment]
@not-an-aardvark Could you go into detail on how that might look? A bit confused since AST selectors focus on nodes, but this rule is really about tokens (specifically operator tokens).

**Comment by not-an-aardvark** [comment]
I was thinking that a user would be able to use an AST selector to select a node that contains an infix operator. For example:

```json
{
  "rules": {
    "space-infix-ops": ["error", {
      "BinaryExpression": "always",
      "AssignmentPattern": "never",
      "BinaryExpression[operator='|'][right.value=0]": "never"
    }]
  }
}
```
